store/src/java/com/zimbra/cs/event/analytics/contact/ContactAnalytics.java

```diff
@@ -1,21 +1,57 @@
 package com.zimbra.cs.event.analytics.contact;
 
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.lucene.analysis.pattern.PatternCaptureGroupFilterFactory;
+
 import com.zimbra.common.calendar.ICalTimeZone;
 import com.zimbra.common.service.ServiceException;
 import com.zimbra.cs.account.Account;
 import com.zimbra.cs.account.Provisioning;
 import com.zimbra.cs.event.EventStore;
 import com.zimbra.cs.mailbox.calendar.Util;
 
-import java.util.List;
-
 public class ContactAnalytics {
     public static enum ContactFrequencyTimeRange {
         LAST_DAY, LAST_WEEK, LAST_MONTH, FOREVER
     }
 
-    public static enum ContactFrequencyGraphTimeRange {
-        CURRENT_MONTH, LAST_SIX_MONTHS, CURRENT_YEAR
+    public static enum ContactFrequencyGraphInterval {
+        DAY("d"), WEEK("w");
+
+        private String shortRepr;
+
+        private ContactFrequencyGraphInterval(String shortRepr) {
+            this.shortRepr = shortRepr;
+        }
+        public static ContactFrequencyGraphInterval of(String str) throws ServiceException {
+            for (ContactFrequencyGraphInterval interval: ContactFrequencyGraphInterval.values()) {
+                if (interval.name().equalsIgnoreCase(str) || interval.shortRepr.equalsIgnoreCase(str)) {
```

Comment:
I'm guessing this accepts "DAY" or "d"?  Is there a reason to accept both?  If so, does that apply to ContactFrequencyGraphTimeRangeUnit too?
My preference would be to keep the accepted values to a minimum - I'd actually prefer to not ignore case too...  That way can keep test cases to a minimum and use simpler code + perhaps use enums in APIs

Reply:
Yeah good point, there's not really any reason for this. I'll keep this as case-sensitive d|w